pkg/apis/clusteroperator/v1alpha1/types.go

```diff
@@ -48,20 +48,37 @@ type ClusterList struct {
 }
 
 type ClusterSpec struct {
-	MasterNodes ClusterNodeGroup `json:"masterNodes"`
+	// MasterNodeGroup specificies the configuration of the master node group
+	MasterNodeGroup ClusterNodeGroup `json:"masterNodeGroup"`
 
+	// ComputeNodeGroups specify the configurations of the compute node groups
 	// +optional
-	ComputeNodeGroups []ClusterNodeGroup `json:"computeNodeGroups,omitempty"`
+	ComputeNodeGroups []ClusterComputeNodeGroup `json:"computeNodeGroups,omitempty"`
 }
 
 type ClusterStatus struct {
+	// MasterNodeGroups is the number of actual master node groups that are
+	// active for the cluster
+	MasterNodeGroups int `json:"masterNodeGroups"`
+
+	// ComputeNodeGroups is the number of actual compute node groups that are
+	// active for the cluster
+	ComputeNodeGroups int `json:"computeNodeGroups"`
```

Comment:
We should rename this at some point as one of these groups will very likely be named "compute" in here. Lots of time to sort that out when we dig into the API though.

Reply:
Sure. From a code standpoint, there is no problem with one of the compute node groups being named "compute". The NodeGroup resource will end up with the name "clustername-compute-compute-somethingtomakeitunique". But if it is going to be confusing to users to have this name ComputeNodeGroups or if that is not the correct term for it, then by all means let's change it.